src/util/files.c

```diff
@@ -102,6 +102,28 @@ char *path_for_fd(int fd) {
     return realpath(fdpath, NULL);
 }
 
+int infer_is_text_plain_utf8(const char *file_path) {
+    /* Infer if the file is valid UTF-8 text */
+    pid_t pid = fork();
+    if (pid < 0) {
+        perror("fork");
+        return 0;
+    }
+    if (pid == 0) {
+        int devnull = open("/dev/null", O_WRONLY);
+        dup2(devnull, STDOUT_FILENO);
+        dup2(devnull, STDERR_FILENO);
+        execlp("iconv", "iconv", "-f", "utf-8", "-t", "utf-8", file_path, NULL);
```

Comment:
Can potentially switch this to use iconv(3) instead or some other UTF-8 validity check.